### PR TITLE
Add Spring Boot auto-configuration for Neo4j vector store

### PR DESCRIPTION
**Pull request by King-Chau**
Add Spring Boot auto-configuration for Neo4j vector store

### CODE REVIEW

**Comment by King-Chau** [comment]
@tzolov Please review, thank you.

**Review by tzolov — CHANGES_REQUESTED** [review]
Thanks @King-Chau 
It looks great! I have few minor recommendations about the code style and one important concern about exposing the  `org.neo4j.*` in the spring-ai API. 

Btw, your PR made me review the other vector stores and I've tried to clean the similar  issues [there](https://github.com/spring-projects/spring-ai/commit/472fadda615a169e2df2b37f1f5b45b15461ec65).

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/neo4j/Neo4jVectorStoreProperties.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import org.springframework.ai.vectorstore.Neo4jVectorStore;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import static org.springframework.ai.autoconfigure.neo4j.Neo4jVectorStoreProperties.CONFIG_PREFIX;
+
+/**
+ * @author Jingzhou Ou
+ */
+@ConfigurationProperties(CONFIG_PREFIX)
```
[the remark this answers is not in the document]
Comment (King-Chau) [reply spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/neo4j/Neo4jVectorStoreProperties.java]: good point, will change

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/neo4j/Neo4jDriverProperties.java [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+
+import static org.springframework.ai.autoconfigure.neo4j.Neo4jDriverProperties.CONFIG_PREFIX;
+
+/**
+ * Properties for Neo4j driver
+ *
+ * @author Jingzhou Ou
+ */
+@ConfigurationProperties(CONFIG_PREFIX)
+public class Neo4jDriverProperties {
```
[the remark this answers is not in the document]
Comment (King-Chau) [reply spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/neo4j/Neo4jDriverProperties.java:31]: good point, will change

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/neo4j/Neo4jDriverProperties.java [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.autoconfigure.neo4j;
+
+import org.neo4j.driver.AuthToken;
```
[the remark this answers is not in the document]
Comment (King-Chau) [reply spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/neo4j/Neo4jDriverProperties.java]: Good point. I already removed org.neo4j API in Neo4jDriverProperties. 
The properties are the same as the existing neo4j spring boot autoconfig.

**Comment by tzolov** [comment]
LGTM, thank you  @King-Chau

**Comment by tzolov** [comment]
Rebased, squashed and merged at: b625ab1c1589dc6be0240b2d76d7023d43a71dcd

**Pull request closed without merging**